### DISCUSSION TITLE
About MCevent data standard format and using SSD.jL for Monte carlo simulation

### DISCUSSION THREAD

**Discussion by Yangjzh0419** [JuliaPhysics/SolidStateDetectors.jl#272] [Q&A] [answered]
Hello, thank you very much for developing such a user-friendly and convenient project , I'm in the process of learning to use it.
At present, I want to use Geant4 simulation results. CSV format data as MCevent for simulate_waveforms(MCevent, sim, δ T = 1u"ns", verbose = false) operation, but the operation will report errors due to its own use problems. Is there a standard. CSV format requirement, and is there a relevant tutorial?
 Thanks again and best wishes!

**Comment by fhagemann** [comment, marked as answer]
Hello, thanks for the feedback and for raising the question!

Does your CSV file look like [this](https://github.com/legend-exp/legend-julia-tutorial/blob/e2567f9515f474bf931c8ba8ee6363b01fbca79a/data/dual-invcoax-mctruth.csv)?
If so, you might want to download the package [LegendTextIO.jl](https://github.com/legend-exp/LegendTextIO.jl) and read in your data using:
```julia
mctruth_filename_csv = "mcevents.csv" # the filename of the CSV file
mc_events = open(read, mctruth_filename_csv, Geant4CSVInput)
```

Please also have a look at [this tutorial](https://github.com/legend-exp/legend-julia-tutorial) which deals in a bit more detail with MC event simulation.

**Reply by Yangjzh0419** [reply]
Hello, the CSV file format I use is just as you have shown. Thank you for your suggestions, which are very helpful to me. I will follow your suggestions to learn.
Thanks again for your help, best wishes!
